java-checks/src/main/java/org/sonar/java/checks/tests/AbstractJUnit5NotCompliantModifierChecker.java

```diff
@@ -36,21 +36,27 @@
 
 public abstract class AbstractJUnit5NotCompliantModifierChecker extends IssuableSubscriptionVisitor {
 
+  public enum ModifierScope {
+    CLASS,
+    CLASS_METHOD,
+    INSTANCE_METHOD
+  }
+
   protected static final String WRONG_MODIFIER_ISSUE_MESSAGE = "Remove this '%s' modifier.";
 
-  protected abstract boolean isNonCompliantModifier(Modifier modifier, boolean isMethod);
+  protected abstract boolean isNonCompliantModifier(Modifier modifier, ModifierScope modifierScope);
 
   protected abstract void raiseIssueOnNonCompliantReturnType(MethodTree methodTree);
 
   protected void raiseIssueOnNonCompliantModifier(ModifierKeywordTree modifier) {
     QuickFixHelper.newIssue(context)
       .forRule(this)
       .onTree(modifier)
-      .withMessage(String.format(WRONG_MODIFIER_ISSUE_MESSAGE, modifier.keyword().text()))
+      .withMessage(WRONG_MODIFIER_ISSUE_MESSAGE, modifier.keyword().text())
       .withQuickFix(() ->
-        JavaQuickFix.newQuickFix("Remove modifier")
-          .addTextEdit(JavaTextEdit.removeTree(modifier))
-          .build())
+        JavaQuickFix.newQuickFix("Remove \"%s\" modifier", modifier.keyword().text())
+          .addTextEdit(JavaTextEdit.removeTextSpan(AnalyzerMessage.textSpanBetween(modifier, true, QuickFixHelper.nextToken(modifier), false)))
```

Comment:
Why do we have to change the way that we generate the quickfix?

Reply:
To have only one `raiseIssueOnNonCompliantModifier` implementation.